test/client/metadataTransfer.test.ts

```diff
@@ -211,51 +235,28 @@ describe('MetadataTransfer', () => {
 
       expect(checkStatus.notCalled).to.be.true;
       expect(result).to.deep.equal({ id: '1' });
+      expect(cancelListenerStub.callCount).to.equal(1);
+      expect(updateListenerStub.callCount).to.equal(0);
     });
 
-    it('should continue polling until cancel operation finishes asynchonously', async () => {
+    it('should exit after checkStatus if transfer is marked for cancelation', async () => {
+      const { checkStatus } = operation.lifecycle;
       const cancelListenerStub = env.stub();
-      const { checkStatus, cancel } = operation.lifecycle;
-      checkStatus.returns({ status: RequestStatus.InProgress });
-      cancel.callsFake(() => {
-        // when cancel is called, set status to canceling
-        checkStatus.returns({ status: RequestStatus.Canceling });
-        return false;
-      });
-
-      const operationPromise = operation.pollStatus();
-
-      queueMicrotask(() => {
-        // cancel right after lifecycle starts
-        operation.cancel();
-        queueMicrotask(() => {
-          // schedule clock to break first wait
-          queueMicrotask(() => {
-            clock.tick(110);
-            queueMicrotask(() => {
-              // schedule clock to break second wait
-              queueMicrotask(() => {
-                clock.tick(110);
-              });
-            });
-          });
-        });
-      });
-
+      const updateListenerStub = env.stub();
       operation.onCancel(() => cancelListenerStub());
-
-      operation.onUpdate((result) => {
-        if (checkStatus.callCount === 2) {
-          // should be canceling by second poll
-          expect(result.status).to.equal(RequestStatus.Canceling);
-          // force third poll to have cancel status
-          checkStatus.returns({ status: RequestStatus.Canceled, done: true });
-        }
+      operation.onUpdate(() => updateListenerStub());
+      checkStatus.onFirstCall().callsFake(async () => {
+        await operation.cancel();
+        return { status: RequestStatus.InProgress, done: false };
       });
 
-      await operationPromise;
+      await operation.start();
+      const result = await operation.pollStatus();
 
+      expect(checkStatus.calledOnce).to.be.true;
+      expect(result).to.deep.equal({ id: '1' });
       expect(cancelListenerStub.callCount).to.equal(1);
+      expect(updateListenerStub.callCount).to.equal(1);
```

Comment:
Ensures checkStatus is only called once since it's canceled within the checkStatus function.  Update and cancel listeners are called once.
The old test isn't really necessary anymore since polling is exited based on the `done` property rather than `status`.  There is coverage of polling stoppage in the metadataDeployApi tests, as the coverage numbers indicate.